app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -1225,12 +1344,208 @@ public void onDateSet(DatePicker view, int year, int month, int day) {
+
+                days = i1;
+                long cal = (1000 * 3600 * 24) * (long) days;
+                reminderDateInMillis = expiryInMillis - cal;
+
+                Date date1 = new Date(reminderDateInMillis);
+                formattedDate = DateFormat.getDateInstance(DateFormat.LONG).format(date1);
+                dateDialogTextView.setText(formattedDate);
+            });
+
+            delay.setTimeInMillis(reminderDateInMillis);
+
+            AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
+            builder.setView(view)
+                    .setTitle(R.string.add_reminder)
+                    .setPositiveButton("Set", new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialogInterface, int i) {
+
+                            Log.i(TAG, "onClick: Set clicked");
+                            DialogFragment dialogFragment = new TimePickerFragment(getContext(), showRemindersLayout, addReminderBtn);
+                            dialogFragment.show(getActivity().getSupportFragmentManager(), "TimePicker");
+                        }
+                    })
+                    .setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialogInterface, int i) {
+                            Log.i(TAG, "onClick: Cancel clicked");
+                        }
+                    });
+            return builder.create();
+        }
+    }
+
+    public static class TimePickerFragment extends DialogFragment implements TimePickerDialog.OnTimeSetListener {
+        Calendar calendar;
+        Context context;
+        View showRemindersLayout;
+        TextView daysTextView, dateReminderView, timeReminderTextView;
+        Button addReminderBtn;
+
+
+        public TimePickerFragment(Context context, View showRemindersLayout, Button addReminderBtn) {
+
+            this.context = context;
+            this.showRemindersLayout = showRemindersLayout;
+            this.addReminderBtn = addReminderBtn;
+
+            daysTextView = showRemindersLayout.findViewById(R.id.daysTextView);
+            dateReminderView = showRemindersLayout.findViewById(R.id.dateReminderView);
+            timeReminderTextView = showRemindersLayout.findViewById(R.id.timeReminderTextView);
+
+        }
+
+        @NonNull
+        @Override
+        public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+
+            calendar = Calendar.getInstance();
+
+            Date date = (Date) timeReminderTextView.getTag();
+
+            if (date != null) {
+                calendar.setTime(date);
+            }
+
+
+            hour = calendar.get(Calendar.HOUR_OF_DAY);
+            minute = calendar.get(Calendar.MINUTE);
+
+            return new TimePickerDialog(getActivity(), this, hour, minute, false);
```

Comment:
I believe this forces it to use AM/PM, which is wrong for some countries. You probably want to use `DateFormat.is24HourFormat(context)` instead of `false` at the end to use the correct system based on the device's language settings.